docs/blog/posts/2024-03-19-splink4.md

```diff
@@ -0,0 +1,108 @@
+
+
+!!! warning
+
+    Splink 3 has entered maintenance mode.  We will continue to apply bugfixes, but new features should be built on the [splink4_dev](https://github.com/moj-analytical-services/splink/tree/splink4_dev) branch.  We are no longer accepting new features on the master (Splink 3) branch.
+
+## Splink 4
+
+The team has been focussing development efforts on [Splink 4](https://github.com/moj-analytical-services/splink/tree/splink4_dev), due to be released later this year.
+
+We’re pleased to announce we’ve recently reached an important milestone: all tests are passing, and all of the tutorial and examples notebooks have been updated and work successfully in the new version
+
+Development releases of Splink 4 have commenced, and you can try it out using `pip install --pre splink`, or try it out in your web browser using the Colab links at the top the [tutorial](https://github.com/moj-analytical-services/splink/tree/splink4_dev/docs/demos/tutorials) and [example](https://github.com/moj-analytical-services/splink/tree/splink4_dev/docs/demos/examples/duckdb) notebooks.
+
+As a result, Splink 3 has entered maintenance mode. We will continue to apply bugfixes, but new features should be built on the [splink4_dev](https://github.com/moj-analytical-services/splink/tree/splink4_dev) branch. We are no longer accepting new features on the master (Splink 3) branch.
+
+### :material-target: Aims of Splink 4
+
+Splink 4 represents an incremental improvement to version 3 that makes Splink easier to use without making any major changes to workflows. The core functionality has not changed - the steps to train a model and predict results are the same, and models trained in Splink 3 will still work in Splink 4.
+
+#### Improve ease of use
+
+The primary aim is to improve the user-facing API so that:
+
+- The user has to write less code to achieve the same result
+- Function imports are simpler and grouped more intuitively
+- Settings and configuration can now be constructed entirely using Python objects, meaning that the user can rely heavily on autocomplete, rather than needing to remember the names of settings.
+- Less dialect-specific code
+
+You can see an example of how the code changes between version 3 and 4 in the screenshot below:
+
+![image](https://gist.github.com/assets/2608005/239eb264-5f79-4db0-a958-d10fdc35d689)
+
+The corresponding code is [here](https://gist.github.com/RobinL/c99712c1fb0b6c80593b5028c0be553a).
+
+#### Improve ease of development
+
+A second important aim of Splink 4 is to improve the internal codebase to make Splink easier to develop for the core team and external contributors. These changes don’t affect the end user, but should enable a faster pace of development.
+
+A wide range of improvements have been made such as:
+
+- Code quality: type hinting, mypy conformance etc.
+- CI much faster
+- Reducing rigidities in dependencies
+- Decoupling parts of the codebase and less mutable state
```

Comment:
I don't really know what mutable state is (and by extension whether this works dramatically). I can't imagine I will be the only one. Is it worth linking off to an explanation somewhere of what this is?

Reply:
Have linked off to a definition